packages/authentication/src/models/auth-user.ts

```diff
@@ -1,11 +1,48 @@
 import { generateEntityId } from "@medusajs/utils"
-import { BeforeCreate, Entity, OnInit, PrimaryKey } from "@mikro-orm/core"
+import {
+  BeforeCreate,
+  Cascade,
+  Entity,
+  ManyToOne,
+  OnInit,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import AuthProvider from "./auth-provider"
+
+type OptionalFields = "password_hash" | "app_metadata" | "user_metadata"
 
 @Entity()
 export default class AuthUser {
+  [OptionalProps]: OptionalFields
+
   @PrimaryKey({ columnType: "text" })
   id!: string
 
+  @Property({ columnType: "text", index: "IDX_auth_user_email" })
+  email: string
```

Comment:
todo: To be discussed, not all providers are email based (e.g steam) see discussion [here](https://github.com/adrien2p/medusa-plugins/pull/120)